packages/jupyter-ai/jupyter_ai/personas/persona_manager.py

```diff
@@ -1,56 +1,92 @@
+from __future__ import annotations
+
 import asyncio
+import os
 from logging import Logger
 from time import time_ns
-from typing import TYPE_CHECKING, ClassVar, Optional
+from typing import TYPE_CHECKING, ClassVar
 
 from importlib_metadata import entry_points
 from jupyterlab_chat.models import Message
 from jupyterlab_chat.ychat import YChat
+from traitlets.config import LoggingConfigurable
 
 from ..config_manager import ConfigManager
 from .base_persona import BasePersona
 
 if TYPE_CHECKING:
     from asyncio import AbstractEventLoop
 
+    from jupyter_server_fileid.manager import (  # type: ignore[import-untyped]
+        BaseFileIdManager,
+    )
+
 # EPG := entry point group
 EPG_NAME = "jupyter_ai.personas"
 
 
-class PersonaManager:
+class PersonaManager(LoggingConfigurable):
     """
     Class that manages all personas for a single chat.
     """
 
     # instance attrs
     ychat: YChat
     config_manager: ConfigManager
-    event_loop: "AbstractEventLoop"
-    log: Logger
+    fileid_manager: BaseFileIdManager
+    root_dir: str
+    event_loop: AbstractEventLoop
+
+    log: Logger  # type: ignore
+    """
+    The `logging.Logger` instance used by this class. This is automatically set
+    by the `LoggingConfigurable` parent class; this declaration only hints the
+    type for type checkers.
+    """
+
     _personas: dict[str, BasePersona]
+    file_id: str
 
     # class attrs
-    _persona_classes: ClassVar[Optional[list[type[BasePersona]]]] = None
+    _persona_classes: ClassVar[list[type[BasePersona]] | None] = None
```

Comment:
I think that for our .jupyter dir use case, this needs to be an instance variable. The reason is that each individual chat file can have different persona classes based on its nearest .jupyter directory.